master/internal/kubernetes/pod.go

```diff
@@ -101,9 +153,191 @@ func (p *pod) startPod(ctx *actor.Context) error {
 	}
 }
 
-func (p *pod) configureResourcesRequirements() v1.ResourceRequirements {
-	return v1.ResourceRequirements{
-		Limits: map[v1.ResourceName]resource.Quantity{
+func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) error {
+	p.pod = msg.updatedPod
+
+	switch msg.updatedPod.Status.Phase {
+	case k8sV1.PodPending:
+		// When pods are deleted, Kubernetes sometimes transitions pod statuses
+		// to pending prior to deleting them. We ignore this state if we have
+		// triggered the deletion.
+		if p.resourcesDeleted {
+			ctx.Log().WithField("pod", p.podName).Info(
```

Comment:
non-blocking: You could add `ctx.AddLabel("pod", p.podName)` to the handler for `PreStart` and the pod name would automatically be added to everything logged by this actor.

Reply:
nice didn't know we had that build in, switched over to using it